moochub-schema.json

```
@@ -1,555 +1,3105 @@
+                                      "example": "https://creativecommons.org/publicdomain/zero/1.0",
+                                      "description": " A generic, general license according to https://spdx.org/licenses/."
+                                    },
+                                    {
+                                      "enum": [
+                                        "proprietary"
+                                      ],
+                                      "example": "proprietary",
+                                      "description": "Marking the course as proprietary."
+                                    }
+                                  ],
+                                  "type": "string"
+                                },
+                                "url": {
+                                  "type": [
+                                    "string",
+                                    "null"
+                                  ],
+                                  "format": "iri",
+                                  "example": null,
+                                  "description": "A link to the specific license document (if any, otherwise set to \"null\" - the identifier pointing at the general license is to be considered the license document then). Has to be \"null\" if identifier is \"proprietary\"."
+                                }
+                              },
+                              "required": [
+                                "identifier"
+                              ]
+                            }
+                          }
+                        },
+                        "required": [
+                          "contentUrl",
+                          "license"
+                        ]
+                      }
+                    },
+                    "required": [
+                      "name",
+                      "type"
+                    ]
+                  }
+                },
+                "required": [
+                  "name",
+                  "type"
+                ]
+              }
+            ]
+          }
+        },
+        "publisher": {
+          "description": "The publisher of the course. Can be a person or an organization.",
+          "anyOf": [
+            {
+              "title": "Organization",
+              "type": "object",
+              "description": "This schema describes the attributes of an organization.",
+              "properties": {
+                "name": {
+                  "type": "string",
+                  "description": "The name of the organization.",
+                  "example": "openHPI"
+                },
+                "identifier": {
+                  "type": [
+                    "string",
+                    "null"
+                  ],
+                  "description": "An identifier for the organization. Can be a link to a homepage, Wikidata,...",
```

```suggestion
                  "description": "An identifier for the organization, e.g. in the form of a link to a homepage.",
```

Links / URLs should not be used as identifiers.

---

Ok, I got what you mean. Nevertheless, AMB (OERSI) is using the id attribute also to store URIs. I think we need to put some more thoughts on this ...
<html><body>
<!--StartFragment-->

title | "License"
-- | --
description | "An object that links to an open source license. Public Domain, Creative Commons, GNU, Apache, MIT and BSD are allowed"
type | "object"
properties |  
id |  
type | "string"
format | "uri"
oneOf |  
0 |  
description | "Creative Commons License or Public Domain"
pattern | "^http[s]?://creativecommons.org/(licenses\|licences\|publicdomain)/.*"
1 |  
description | "GNU License"
pattern | "^http[s]?://www.gnu.org/licenses/.*"
2 |  
description | "Apache License"
pattern | "^http[s]?://www.apache.org/licenses/.*"

<!--EndFragment-->
</body>
</html>

---

should be solved with new version